src/odemis/acq/move.py

```diff
@@ -194,125 +294,221 @@ def _doCryoSwitchSamplePosition(future, stage, focus, align, target):
+                for component, sub_move in sub_moves:
+                    run_sub_move(future, component, sub_move)
+            except IndexError:
+                # In case the required movement is invalid/unreachable with the smaract 5dof stage
+                # Move all linear axes first then rotational ones using the fallback_submoves
+                logging.debug("This move {} is unreachable, trying to move all axes at once...".format(sub_move))
+                for component, sub_move in fallback_submoves:
+                    run_sub_move(future, component, sub_move)
+
+        except CancelledError:
+            logging.info("_doCryoLoadSample cancelled.")
+        except Exception as exp:
+            logging.exception("Failure to move to {} position.".format(target_pos_str.get(target, lambda: "unknown")))
+            raise
+        finally:
+            with future._task_lock:
+                if future._task_state == CANCELLED:
+                    raise CancelledError()
+                future._task_state = FINISHED
+
+    elif role == "meteor":
         try:
-            for component, sub_move in sub_moves:
-                run_sub_move(future, component, sub_move)
-        except IndexError:
-            # In case the required movement is invalid/unreachable with the smaract 5dof stage
-            # Move all linear axes first then rotational ones using the fallback_submoves
-            logging.debug("This move {} is unreachable, trying to move all axes at once...".format(sub_move))
-            for component, sub_move in fallback_submoves:
-                run_sub_move(future, component, sub_move)
-
-    except CancelledError:
-        logging.info("_doCryoLoadSample cancelled.")
-    except Exception as exp:
-        logging.exception("Failure to move to {} position.".format(target_pos_str.get(target, lambda: "unknown")))
-        raise
-    finally:
-        with future._task_lock:
-            if future._task_state == CANCELLED:
-                raise CancelledError()
-            future._task_state = FINISHED
+            # get the meta data 
+            focus_md = focus.getMetadata()
+            focus_deactive = focus_md[model.MD_FAV_POS_DEACTIVE]
+            focus_active = focus_md[model.MD_FAV_POS_ACTIVE]
+            # To hold the ordered sub moves list
+            sub_moves = []
+            # Create axis->pos dict from target position given smaller number of axes
+            filter_dict = lambda keys, d: {key: d[key] for key in keys}
+            # get the current label 
+            current_pos_label = getCurrentPositionLabel(stage.position.value, stage)
+            # get the set point position
+            target_pos = getTargetPosition(target, stage)
+            # determine the order of moves for the stage and focuser
+            if current_pos_label in [LOADING, SEM_IMAGING]:
+                if target in [SEM_IMAGING, GRID_1]:  
+                    # TODO put the correct order of moves 
```

Comment:
If we are doing a loading to sem move, then we'll need to make sure the focus is parked.